lib/engine/game/base.rb

```diff
@@ -856,6 +856,18 @@ def two_player?
         @two_player ||= @players.size == 2
       end
 
+      def add_extra_tile(tile)
+        game_error('Add extra tile only works if unlimited') unless tile.unlimited
+
+        # Find the highest tile that exists of this type in the tile list and duplicate it.
```

Comment:
An alternative would be to make the find-max code part of the dup method. When you would guarantee that dup isn't called with a non max tile.